### ISSUE TITLE
including breaks my build due to ambiguous compile key

### ISSUE THREAD

**Issue by forestofrain** [sbt/sbt-protobuf#37]
I can't include this plugin in my build because it breaks it. Here is the error I get when adding `addSbtPlugin("com.github.gseitz" % "sbt-protobuf" % "0.4.0")` to `plugins.sbt`:

```
error: reference to compile is ambiguous;
it is imported twice in the same scope by
import _root_.sbtprotobuf.ProtobufPlugin._
and import Keys._
  settings(mainClass in compile := Some("whatever")
```

I found a similar issue from [twitter scrooge](https://github.com/twitter/scrooge/issues/209). If i follow @matthandlersux workaround `import Keys.{compile => sbtCompile}` and change all my `mainClass in compile` to `mainClass in sbtCompile`, then adding the plugin works.

https://github.com/trueaccord/ScalaPB depends upon this plugin, and it also breaks my build.

**Comment by monzonj** [comment]
This is not an autoplugin.

You should import:
```
import sbtprotobuf.{ProtobufPlugin=>PB}
```

instead of
```
import _root_.sbtprotobuf.ProtobufPlugin._
```

and explicitly add the settings:
```
settings = Defaults.defaultSettings ++ PB.protobufSettings
````

**Comment by forestofrain** [comment]
Simply adding the plugin to `plugins.sbt` breaks my multi project build.

The error I pasted is output from SBT. And this happened as soon as I simply added the plugin, never including any settings in my build file. I never wrote this line of code in my build file:
```
import _root_.sbtprotobuf.ProtobufPlugin._
```

Anyhow, I have moved away from protobuf. Wanted to capture this for any others that might see the same issue. If you follow the workaround of aliasing compile `import Keys.{compile => sbtCompile}` in your build, then everything works as expected. If you don't use `compile` in your build file, then you won't see the issue. The issue that was linked from scrooge shows a probable fix.

**Comment by matthandlersux** [comment]
for convenience, here's a link to how this was fixed for scrooge: https://github.com/twitter/scrooge/pull/210/files#diff-3d1ecf501419eb7452297ced4c03b47eR13

**Comment by jroper** [comment]
I'm seeing this too, the import is added automatically by sbt when you use `*.sbt` files, and it's a bug in (or feature of) the Scala compiler:

https://issues.scala-lang.org/browse/SI-2133

Even though `ProtobufPlugin.compile` is private, it's being imported.  Since this is never going to be fixed in Scala 2.10, it should be fixed here, there are three ways to fix it, one is to rename `compile`, another is to move it into a different object, and finally it could be converted to an auto plugin where automatic imports to sbt files are explicitly controlled.

**Comment by xuwei-k** [comment]
Thanks all. I just fixed and released `0.5.2`
